### PR TITLE
MLH-503 TC-068-087

### PR DESCRIPTION
**Pull request by MariaMarkel**

### CODE REVIEW

File: test/smoke/createButton.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+const sel = require ('../../data/selectors.json');
+const exp = require ('../../data/expected.json');
+const data = require ('../../data/testData.json');
+const inputValues4 = require ('../../helpers/inputValues4');
+import { expect } from 'chai';
+
+describe('Functionality of Create Button', function () {
+
+    describe('Happy path', function () {
+
```
Comment (vasylynash) [inline test/smoke/createButton.js:10]: Remove one empty line

File: test/smoke/createButton.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        });
+
+        it('TC-070  The button is active when all required fields are selected', function () {
+            browser.url('');
+            inputValues4(data.name, data.gender.she, data.age, data.storyType.Comedy)
+            const create = $(sel.create).isEnabled();
+            browser.pause(3000);
+            expect(create).to.be.equal(true);
+        });
+
```
Comment (vasylynash) [inline test/smoke/createButton.js:32]: Remove empty line

File: test/smoke/createButtonNegative.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
+const sel = require ('../../data/selectors.json');
+const data = require ('../../data/testData.json');
+const inputValues4 = require ('../../helpers/inputValues4');
+const imageUpload = require('../../helpers/imageUpload.js');
+import { expect } from 'chai';
+
+describe('Create Button stays inactive before options 1-4 are selected', function () {
+
+    describe('Happy path', function () {
+
```
Comment (vasylynash) [inline test/smoke/createButtonNegative.js:10]: Remove one empty line

File: test/smoke/createButtonNegative.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        it('TC-082  The submit button is disabled when only 2 options are selected: gender She and Comedy story type', function () {
+            browser.url('');
+            $$(sel.gender)[data.gender.she].click();
+            $(sel.storyClick).click();
+            $$(sel.storyType)[data.storyType.Comedy].click();
+            const create = $(sel.create).isEnabled();
+            browser.pause(3000);
+            expect(create).to.be.equal(false);
+        });
+
```
Comment (vasylynash) [inline test/smoke/createButtonNegative.js:125]: Remove empty line

File: test/smoke/modifySelections.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+const sel = require ('../../data/selectors.json');
+const data = require ('../../data/testData.json');
+const inputValues4 = require ('../../helpers/inputValues4');
+import { expect } from 'chai';
+
+describe('The chosen selections can be modified any time before clicking the \'Create!\' button', function () {
+
+    describe('Happy path', function () {
+
```
Comment (vasylynash) [inline test/smoke/modifySelections.js:9]: Remove one empty line

File: test/smoke/modifySelections.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+            expect(create).to.be.equal(true);
+        });
+
+        it('TC-083.001  All previous selections are modified, the button is active', function () {
+            inputValues4(data.name, data.gender.he, data.age, data.storyType.JourneyAndReturn)
+            const create = $(sel.create).isEnabled();
+            browser.pause(3000);
+            expect(create).to.be.equal(true);
+        });
+
```
Comment (vasylynash) [inline test/smoke/modifySelections.js:25]: Remove empty line

File: test/smoke/storyPage.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+const inputValues4 = require ('../../helpers/inputValues4');
+const inputValues4andClick = require ('../../helpers/inputValues4andClick');
+const imageUpload = require ('../../helpers/imageUpload');
+const path = require ('path');
+import { expect } from 'chai';
+
+ describe('The story appears on the same page and has 3 (4 if an image was selected) main elements', function () {
+
+    describe('The story page has 3 main elements: the page header, a story and a \'Try again !\' button.', function () {
+
```
Comment (vasylynash) [inline test/smoke/storyPage.js:12]: Remove empty line

File: test/smoke/storyPage.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        it('TC-084.002 No image is displayed on the story page', function () {
+                const storyImage = $(sel.storyImage).isDisplayed();
+                expect(storyImage).to.be.equal(false);
+            });
+
+        it('TC-084.003 Story is displayed', function () {
+            const storyText = $(sel.storyText).isDisplayed();
+            expect(storyText).to.be.equal(true);
+        });
+
```
Comment (vasylynash) [inline test/smoke/storyPage.js:37]: Remove empty line

File: test/smoke/storyPage.js [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+        it('TC-085  The story appears on the same page', function () {
+            browser.url('');
+            const appUrl = browser.getUrl();
+            inputValues4andClick(data.name, data.gender.she, data.age, data.storyType.Comedy)
+            browser.pause(3000);
+            const storyUrl = browser.getUrl();
+            expect(appUrl).to.be.equal(storyUrl);
+        });
+
```
Comment (vasylynash) [inline test/smoke/storyPage.js:47]: Remove empty line

File: test/smoke/storyPage.js [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+        it('TC-086  Happy path to story page: Try again button exists' , function () {
+            browser.url('');
+            inputValues4(data.name, data.gender.she, data.age, data.storyType.Comedy)
+            imageUpload()
+            $(sel.create).click();
+            const tryAgain = $(sel.tryAgain).isDisplayed();
+            browser.pause(3000);
+            expect(tryAgain).to.be.equal(true);
+
```
Comment (vasylynash) [inline test/smoke/storyPage.js:61]: Remove empty line

File: test/smoke/storyPage.js [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+        it('TC-086.001 Story header is displayed', function () {
+            const storyHeader = $(sel.storyHeader).isDisplayed();
+            expect(storyHeader).to.be.equal(true);
+        });
+
+        it('TC-086.002 An image is displayed on the story page', function () {
+                const storyImage = $(sel.storyImage).isDisplayed();
+                expect(storyImage).to.be.equal(true);
+            });
```
Comment (vasylynash) [inline test/smoke/storyPage.js:72]: Fix the parenthesis alignment

File: test/smoke/storyToRead.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
+const sel = require ('../../data/selectors.json');
+const exp = require ('../../data/expected.json');
+const data = require ('../../data/testData.json');
+const inputValues4andClick = require ('../../helpers/inputValues4andClick');
+import { expect } from 'chai';
+
+describe('Each type of the story uses the correct template', function () {
+
+    describe('Happy path', function () {
+
```
Comment (vasylynash) [inline test/smoke/storyToRead.js:10]: Remove empty line

File: test/smoke/storyToRead.js [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+
+        it('TC-215 The Comedy type of the story uses the correct template', function () {
+            browser.url('');
+            inputValues4andClick(data.name, data.gender.she, data.age, data.storyType.Comedy)
+
```
Comment (vasylynash) [inline test/smoke/storyToRead.js:15]: Remove empty line

File: test/smoke/storyToRead.js [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+            const text = $(sel.create).getText();
+            expect(text).to.be.equal(exp.labelCreate);
+        });
+
```
Comment (vasylynash) [inline test/smoke/storyToRead.js:19]: Remove empty line

File: helpers/imageUpload.js [hunk 15]
```diff
@@ -7,7 +7,7 @@ function imageUpload(){
         fileUpload
     );
     fileUpload.waitForDisplayed();
-    const filePath = path.join(__dirname, "../data/123.jpeg");
+    const filePath = path.join(__dirname, "../data/Bug.jpg");
```
Comment (vasylynash) [inline helpers/imageUpload.js:10]: The function should not use the hardcoded path

**Review by vasylynash — CHANGES_REQUESTED** [review]
Please make the data files consistent with the master (resolve conflicts in selectors, test data, expected, etc.). Make sure not to erase the data that already exists in the master branch.

File: helpers/imageUpload.js (hunk 15 above)
Comment (MariaMarkel) [reply helpers/imageUpload.js:10]: need more clarification please: if i have to change the path here, how do i do that?

**Review by Anastasiia-Bandurova — APPROVED** [review]

**Review by vasylynash — APPROVED** [review]

**Pull request merged**